Proposed fix for building on macOS < 10.9,
combined with backport of nearby fix for https://core.tcl-lang.org/tk/info/0625d100

Upstream-Status: Submitted [https://core.tcl-lang.org/tk/info/d3d7cbc6]

--- macosx/tkMacOSXMenu.c
+++ macosx/tkMacOSXMenu.c
@@ -1002,26 +1002,32 @@ TkpPostMenu(
         inPostMenu = false;
         return result;
     }
-    if (itemIndex >= numItems) {
-    	itemIndex = numItems - 1;
-    }
-    if (itemIndex >= 0) {
-	item = [menu itemAtIndex:itemIndex];
-    }
-
     /*
      * The post commands could have deleted the menu, which means we are dead
      * and should go away.
      */
-
     if (menuPtr->tkwin == NULL) {
     	return TCL_OK;
     }
 
+    if (itemIndex >= numItems) {
+	itemIndex = numItems - 1;
+    }
+    if (itemIndex >= 0) {
+	item = [menu itemAtIndex:itemIndex];
+    }
+
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
+    if (@available(macOS 10.14, *)) {
+	[menu popUpMenuPositioningItem:item
+			    atLocation:location
+				inView:nil
+			    appearance:realWinView.effectiveAppearance];
+    } else
+#endif
     [menu popUpMenuPositioningItem:item
 			atLocation:location
-			    inView:nil
-			appearance:realWinView.effectiveAppearance];
+			    inView:nil];
     inPostMenu = false;
     return TCL_OK;
 }
--- macosx/tkMacOSXPrivate.h
+++ macosx/tkMacOSXPrivate.h
@@ -488,13 +488,15 @@
 - (NSMenuItem *)itemInSupermenu;
 @end
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
 // Need undocumented appearance: argument
 @interface NSMenu(TKMenu)
 - (BOOL)popUpMenuPositioningItem:(NSMenuItem *)item
 		      atLocation:(NSPoint)location
 			  inView:(NSView *)view
 		      appearance:(NSAppearance *)appearance;
 @end
+#endif
 
 @interface NSMenuItem(TKUtils)
 + (id)itemWithSubmenu:(NSMenu *)submenu;
